engine/src/main/scala/cromwell/engine/io/nio/NioFlow.scala

```diff
@@ -109,7 +110,12 @@ class NioFlow(parallelism: Int,
 
   private def hash(hash: IoHashCommand): IO[String] = {
     hash.file match {
-      case gcsPath: GcsPath => IO { gcsPath.cloudStorage.get(gcsPath.blob).getCrc32c }
+      case gcsPath: GcsPath => IO {
+        logger.info(s"Going to execute hash command for GCS file ${gcsPath.pathAsString}")
```

Comment:
I liked the concept above (and in Khalid's PR) of tying these together with an ad-hoc UUID or hash so we can match together the start-end logs (eg if they're all hashing the same reference file it's good to know which start time links to which end time)

Reply:
I'm almost confident that in our case GcsNioFlow will be used instead of this one, but anyway I added uuid here

Reply:
yeah that's what I thought too but Khalid was seeing some surprising things in his PR...